extension/src/prototypes/view/ui-containers/AppHeader.tsx

```diff
@@ -14,14 +15,34 @@ import { MainMenuButton } from "./MainMenuButton";
 import { SettingsButton } from "./SettingsButton";
 import { ToolButtons } from "./ToolButtons";
 
-export const AppHeader: FC = () => {
+type Props = {
+  arURL?: string;
+};
+
+export const AppHeader: FC<Props> = ({ arURL }) => {
   const hidden = useAtomValue(hideAppOverlayAtom);
   const [showShareModal, setShowShareModal] = useAtom(showShareModalAtom);
   const theme = useTheme();
   const isMobile = useMediaQuery(theme.breakpoints.down("mobile"));
+  const getRootLayers = useAtomValue(rootLayersAtom);
+  const [arSharedLayers, setArSharedLayers] = useState<string | undefined>();
+
+  useEffect(() => {
+    (async () => {
+      if (!getRootLayers || getRootLayers?.length === 0 || !arURL) return;
+      const layers = getRootLayers.map(({ type, id }) => ({
+        type,
+        id,
+      }));
```

Comment:
We should pass the `id`, `currentDataIdAtom` and type. So we need to make an atom to get the currentDataIdAtom.
```
const rootLayersForAR = useMemo(() => atom((get) => {
  const rootLayers = get(rootLayersAtom);
  return rootLayers.map(({ type, id, currentDataIdAtom }) => ({
    type,
    datasetId: id,
    dataId: get(currentDataIdAtom),
  }))
}), []);
```

Reply:
Thanks for this. Fixed :heart: